ui/src/app/job-list/job-list.component.ts

```diff
@@ -66,15 +67,19 @@ export class JobListComponent implements OnInit {
       if (event instanceof NavigationError) {
         this.handleError(event.error);
       }
+      if (event instanceof NavigationEnd && this.jobStream.value.needsRefresh) {
+        this.reloadJobs(this.route.snapshot.queryParams['q'], false);
+      }
     });
   }
 
-  reloadJobs(query: string) {
+  reloadJobs(query: string, displayLoading = true): void {
```

Comment:
opt: it would be nice if we had the less invasive "snackbar" loading indicator in the false case here i.e. give the user some indication that something is happening so that they aren't surprised when the UI shuffles around